iroh/src/node.rs

```diff
@@ -1480,6 +1481,77 @@ impl<D: BaoStore> RpcHandler<D> {
         rx.into_stream()
     }
 
+    fn blob_read_at(
+        self,
+        req: BlobReadAtRequest,
+    ) -> impl Stream<Item = RpcResult<BlobReadAtResponse>> + Send + 'static {
+        let (tx, rx) = flume::bounded(RPC_BLOB_GET_CHANNEL_CAP);
+        let entry = self.inner.db.get(&req.hash);
+        self.inner.rt.spawn_pinned(move || async move {
+            if let Err(err) = read_loop(
+                req.offset,
+                req.len,
+                entry,
+                tx.clone(),
+                RPC_BLOB_GET_CHUNK_SIZE,
+            )
+            .await
+            {
+                tx.send_async(RpcResult::Err(err.into())).await.ok();
+            }
+        });
+
+        async fn read_loop<M: Map>(
+            offset: u64,
+            len: usize,
+            entry: Option<impl MapEntry<M>>,
+            tx: flume::Sender<RpcResult<BlobReadAtResponse>>,
+            max_chunk_size: usize,
+        ) -> anyhow::Result<()> {
+            let entry = entry.ok_or_else(|| anyhow!("Blob not found"))?;
+            let size = entry.size();
+            tx.send_async(Ok(BlobReadAtResponse::Entry {
+                size,
+                is_complete: entry.is_complete(),
+            }))
+            .await?;
+            let mut reader = entry.data_reader().await?;
+
+            let (num_chunks, chunk_size) = if len <= max_chunk_size {
+                (1, len)
+            } else {
+                (
+                    (len as f64 / max_chunk_size as f64).ceil() as usize,
```

Comment:
weird to use f64 here... It is shorter than the integer variant, but will fail if len or max_chunk_size is > 2^53.

Reply:
how would I round up otherwise?

Reply:
```rust
len / max_chunk_size + if len % max_chunk_size == 0 { 0 } else { 1 }
```

or

```rust
len / max_chunk_size + (len % max_chunk_size !=0) as usize
```

or something. With a bit of luck the optimizer will do only 1 integer division for both division and rest.

Reply:
Note that I don't worry about speed here. It's just weird to use fp for integer ops, and it will fail in weird ways as soon as you exceed the range of integers that can be losslessly represented in a fp num.

Reply:
ah yes, that is of course nice